src/frontend/Komponenter/Personoversikt/Vedtaksperioder.tsx

```diff
@@ -49,31 +52,62 @@ const endring = (endring?: AndelHistorikkEndring) =>
         </Link>
     );
 
-const historikkRad = (andel: AndelHistorikk) => (
-    <Rad type={andel.endring?.type}>
-        <td>
-            {formaterNullableMånedÅr(andel.andel.stønadFra)}
-            {' - '}
-            {formaterNullableMånedÅr(andel.andel.stønadTil)}
-        </td>
-        <td>{formaterTallMedTusenSkille(andel.andel.inntekt)}</td>
-        <td>{formaterTallMedTusenSkille(andel.andel.beløp)}</td>
-        <td>{vedtakstidspunkt(andel)}</td>
-        <td>{andel.saksbehandler}</td>
-        <td>{endring(andel.endring)}</td>
-    </Rad>
-);
+const etikettType = (periodeType: EPeriodetype) => {
+    switch (periodeType) {
+        case EPeriodetype.HOVEDPERIODE:
+            return 'suksess';
+        case EPeriodetype.PERIODE_FØR_FØDSEL:
+            return 'info';
+        case EPeriodetype.UTVIDELSE:
+            return 'fokus';
+        case EPeriodetype.FORLENGELSE:
+            return 'advarsel';
+    }
+};
+
+const historikkRad = (andel: AndelHistorikk) => {
+    return (
+        <Rad type={andel.endring?.type}>
+            <td>
+                {formaterNullableMånedÅr(andel.andel.stønadFra)}
+                {' - '}
+                {formaterNullableMånedÅr(andel.andel.stønadTil)}
+            </td>
+            <td>
+                <EtikettBase mini type={etikettType(andel.periodeType)}>
+                    {storForbokstavOgRestenSmå(andel.periodeType)}
+                </EtikettBase>
+            </td>
+            <td>{andel.aktivitet}</td>
+            <td>{formaterTallMedTusenSkille(andel.andel.inntekt)}</td>
+            <td>{andel.andel.samordningsfradrag}</td>
+            <td>{formaterTallMedTusenSkille(andel.andel.beløp)}</td>
+            <td>{vedtakstidspunkt(andel)}</td>
+            <td>{andel.saksbehandler}</td>
+            <td>
+                <Link className="lenke" to={{ pathname: `/behandling/${andel.behandlingId}` }}>
+                    {storForbokstavOgRestenSmå(andel.behandlingType)}
```

Comment:
Her tror jeg det er bedre å bruke `behandlingstypeTilTekst[andel.behandlingType]` for at teksten skal være mer presis enn det enum-verdien er?

Reply:
Fikset i 325f76b